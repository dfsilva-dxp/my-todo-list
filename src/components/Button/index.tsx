import * as S from "./styles";

import { IButtonProps } from "./types";

const Button = ({ children, icon }: IButtonProps) => {
  return (
    <S.ButtonContent>
      {!!children && children}
      {!!icon && icon}
    </S.ButtonContent>
  );
};

export default Button;
