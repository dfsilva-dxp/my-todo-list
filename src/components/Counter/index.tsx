import * as S from "./styles";

import { ICounterProps } from "./types";

const Counter = ({ value }: ICounterProps) => {
  return (
    <S.Wrapper role="badge" aria-label="counter">
      {value}
    </S.Wrapper>
  );
};

export default Counter;
