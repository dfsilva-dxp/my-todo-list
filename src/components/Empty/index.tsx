import clipboardSVG from "@assets/clipboard.svg";

import * as S from "./styles";

const Empty = () => {
  return (
    <S.Wrapper>
      <img
        src={clipboardSVG}
        alt="clipboard icon"
        title="clipboard icon"
        aria-label="clipboard icon"
      />

      <S.TextEmptyWrapper>
        <h4>Você ainda não tem tarefas cadastradas</h4>
        <p>Crie tarefas e organize seus itens a fazer</p>
      </S.TextEmptyWrapper>
    </S.Wrapper>
  );
};

export default Empty;
