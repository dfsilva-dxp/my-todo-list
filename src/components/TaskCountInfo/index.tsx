import * as S from "./styles";

import { ITaskCountInfoProps } from "./types";

const TaskCountInfo = ({
  children,
  badge,
  variant = "created"
}: ITaskCountInfoProps) => {
  return (
    <S.Wrapper variant={variant} role="info" aria-label="count info">
      <p>{children}</p>
      {badge}
    </S.Wrapper>
  );
};

export default TaskCountInfo;
