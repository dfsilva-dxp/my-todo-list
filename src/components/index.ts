export { default as Button } from "./Button";
export { default as Checkbox } from "./Checkbox";
export { default as Container } from "./Container";
export { default as Counter } from "./Counter";
export { default as Empty } from "./Empty";
export { default as Form } from "./Form";
export { default as Header } from "./Header";
export { default as Input } from "./Input";
export { default as Logo } from "./Logo";
export { default as Task } from "./Task";
export { default as TaskCountInfo } from "./TaskCountInfo";
export { default as TaskList } from "./TaskList";
