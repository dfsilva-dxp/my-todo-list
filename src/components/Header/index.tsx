import { Logo } from "@/components";

import * as S from "./styles";

import logoSVG from "@/assets/logo.svg";

const Header = () => {
  return (
    <S.Wrapper role="banner">
      <Logo
        src={logoSVG}
        title="Logo Todo App"
        alt="Logo Todo App"
        aria-label="logo"
      />
    </S.Wrapper>
  );
};

export default Header;
