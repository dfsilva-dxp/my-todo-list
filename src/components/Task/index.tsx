import { useState } from "react";
import { Trash } from "phosphor-react";

import { Button, Checkbox } from "@/components";

import useTaskContext from "@/hooks/useTasksContext";

import * as S from "./styles";

import { ITaskProps } from "./types";

const Task = ({ description, checked, taskId }: ITaskProps) => {
  const [isChecked, setIsChecked] = useState(checked);

  const { updateTask } = useTaskContext();

  const onCheck = (status: boolean) => {
    setIsChecked(status);
    updateTask(taskId, status);
  };

  return (
    <S.Wrapper role="listitem" aria-label="task card">
      <Checkbox isChecked={isChecked} onCheck={onCheck} />

      <S.TaskDescription status={isChecked ? "done" : "created"}>
        {description}
      </S.TaskDescription>

      <Button
        type="button"
        minimal
        size="small"
        icon={<Trash size={20} weight="bold" />}
      />
    </S.Wrapper>
  );
};

export default Task;
