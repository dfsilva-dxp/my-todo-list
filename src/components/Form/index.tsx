import { FormEvent, useState } from "react";
import { PlusCircle } from "phosphor-react";

import { Button, TextField } from "@/components";

import useTaskContext from "@/hooks/useTasksContext";

import * as S from "./styles";

import { ITask } from "@components/Task/types";

const Form = () => {
  const [inputValue, setInputValue] = useState("");

  const { createTask } = useTaskContext();

  const onInputChange = (value: string) => {
    setInputValue(value);
  };

  const handleSubmit = (evt: FormEvent<HTMLFormElement>) => {
    evt.preventDefault();
    const id = Math.floor(Date.now() * Math.random());

    const data: ITask = {
      id: String(id),
      description: inputValue,
      status: "created",
      checked: false,
      createdAt: new Date()
    };

    createTask(data);
    setInputValue("");
  };

  return (
    <S.Wrapper onSubmit={handleSubmit}>
      <TextField
        placeholder="Adicione uma nova tarefa"
        initialValue={inputValue}
        onInputValue={onInputChange}
      />
      <Button
        type="submit"
        size="small"
        icon={<PlusCircle size={20} weight="bold" />}
      >
        Criar
      </Button>
    </S.Wrapper>
  );
};

export default Form;
