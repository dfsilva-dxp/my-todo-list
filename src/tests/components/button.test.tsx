import { renderWithTheme } from "@/utils/tests/renderWithTheme";
import { screen } from "@testing-library/react";
import { PlusCircle } from "phosphor-react";

import { Button } from "@/components";

describe("<Button />", () => {
  it("should render the medium button by default", () => {
    const { container } = renderWithTheme(<Button>Buy now</Button>);
    expect(screen.getByRole("button", { name: /Buy now/i })).toHaveStyle({
      height: "4rem",
      padding: "0.8rem 3.2rem",
      "font-size": "1rem"
    });
    expect(container.firstChild).toMatchSnapshot();
  });

  it("should render the small button when size small is passed", () => {
    renderWithTheme(<Button size="small">Buy now</Button>);
    expect(screen.getByRole("button", { name: /Buy now/i })).toHaveStyle({
      height: "3rem",
      "font-size": "0.875rem"
    });
  });

  it("should render the large button when size large is passed", () => {
    renderWithTheme(<Button size="large">Buy now</Button>);
    expect(screen.getByRole("button", { name: /Buy now/i })).toHaveStyle({
      height: "5rem",
      padding: "0.8rem 4.8rem",
      "font-size": "1.2rem"
    });
  });

  it("should render the full width button when fullWidth option is passed", () => {
    renderWithTheme(<Button fullWidth>Buy now</Button>);
    expect(screen.getByRole("button", { name: /Buy now/i })).toHaveStyle({
      width: "100%"
    });
  });

  it("should render an icon", () => {
    renderWithTheme(
      <Button icon={<PlusCircle size={20} weight="bold" data-testid="icon" />}>
        Buy now
      </Button>
    );
    expect(screen.getByText(/Buy now/i)).toBeInTheDocument();
    expect(screen.getByTestId("icon")).toBeInTheDocument();
  });

  it("should render a minimal", () => {
    renderWithTheme(
      <Button
        icon={<PlusCircle size={20} weight="bold" data-testid="icon" />}
        minimal
      >
        Buy now
      </Button>
    );

    expect(screen.getByRole("button", { name: /buy now/i })).toHaveStyle({
      background: "none"
    });

    expect(screen.getByRole("button", { name: /buy now/i })).toHaveStyleRule(
      "background",
      "#0D0D0D",
      {
        modifier: ":hover"
      }
    );
  });

  it("should render a disabled Button", () => {
    renderWithTheme(<Button disabled>Buy now</Button>);

    expect(screen.getByRole("button", { name: /buy now/i })).toHaveStyleRule(
      "cursor",
      "not-allowed",
      {
        modifier: ":disabled"
      }
    );
  });

  it("should render button as a link", () => {
    renderWithTheme(
      <Button as="a" href="/link">
        Buy now
      </Button>
    );

    expect(screen.getByRole("link", { name: /buy now/i })).toHaveAttribute(
      "href",
      "/link"
    );
  });
});
