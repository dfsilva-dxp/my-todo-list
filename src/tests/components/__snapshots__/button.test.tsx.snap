// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should render the medium button by default 1`] = `
.c0 {
  background: #4EA8DE;
  padding: 0.5rem;
  color: #F2F2F2;
  font-weight: 700;
  border: 0;
  border-radius: 0.5rem;
  cursor: pointer;
  -webkit-text-decoration: none;
  text-decoration: none;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  gap: 0.5rem;
  height: 4rem;
  padding: 0.8rem 3.2rem;
  font-size: 1rem;
}

.c0:hover {
  background: #1E6F9F;
}

<button
  class="c0"
>
  <span>
    Buy now
  </span>
</button>
`;
